airflow/ui/src/utils/query.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import type { Query } from "@tanstack/react-query";
+
+import { useDagServiceGetDagDetails } from "openapi/queries";
+import type { TaskInstanceState } from "openapi/requests/types.gen";
+import { useConfig } from "src/queries/useConfig";
+
+export const isStatePending = (state?: TaskInstanceState | null) =>
+  state === "deferred" ||
+  state === "scheduled" ||
+  state === "running" ||
+  state === "up_for_reschedule" ||
+  state === "up_for_retry" ||
+  state === "queued" ||
+  state === "restarting" ||
+  !Boolean(state);
+
+export type PartialQueryKey = { baseKey: string; options?: Record<string, unknown> };
+
+export const doQueriesMatch = (query: Query, queriesToMatch: Array<PartialQueryKey>) => {
+  const [baseKey, options] = query.queryKey;
+
+  const matchedKey = queriesToMatch.find((qk) => qk.baseKey === baseKey);
+
+  if (!matchedKey) {
+    return false;
+  }
+
+  return matchedKey.options
+    ? Object.entries(matchedKey.options).every(
+        ([key, value]) => typeof options === "object" && (options as Record<string, unknown>)[key] === value,
+      )
+    : true;
+};
```

Comment:
I'm not sure I get this piece.

Can you explain what this is achieving on top of the invalidation API from react-query.

For instance this kind of matching can be done with the auto-generated functions: 
```
{ baseKey: useDagRunServiceGetDagRunsKey, options: { dagIds: [dagId] } },
{ baseKey: useTaskInstanceServiceGetTaskInstancesKey, options: { dagId, dagRunId: "~" }
```

Reply:
Invalidating groups for multiple ids ?

Reply:
Our autogenerated keys put all of the parameters into an object, as so `[ "generalKey" , {...specificParams}]`. I only have two options for query invalidation. Invalidate everything with "generalKey" or invalidate whatever exactly matches the specific parameters. Which means our query invalidation needs to know orderBy, limit, offset, taskId, etc.

This matching function will go through and only check against the parameters you care about and ignore the rest.

Reply:
I got it, great!